docs/static/persistent-queues.asciidoc

```diff
@@ -1,44 +1,228 @@
+queue will be sized at the value of `queue.max_bytes` specified in
+`logstash.yml`. 
+The default is 1024mb (1gb).
++
+Be sure that your disk has sufficient capacity to handle the cumulative total of `queue.max_bytes` across all persistent queues.
+The total of `queue.max.bytes` for _all_ queues should be
+lower than the capacity of your disk. 
++
+TIP: If you are using persistent queues to protect against data loss, but don't
+require much buffering, you can set `queue.max_bytes` to a smaller value, such
+as 10mb.
+A smaller value produces smaller queues and improves queue performance. 
+
+`queue.checkpoint.acks`:: Sets the number of acked events before forcing a checkpoint. 
+Default is `1024`. Set to `0` for unlimited.
+`queue.checkpoint.writes`:: Sets the maximum number of written events before a forced checkpoint. 
+Default is `1024`. Set to `0` for unlimited.
++
+To avoid losing data in the persistent queue, you can set `queue.checkpoint.writes: 1` to force a checkpoint after each event is
+written. Keep in mind that disk writes have a resource cost. Setting this value
+to `1` ensures maximum durability, but can severely impact performance.
+See <<durability-persistent-queues>> to better understand the trade-offs.
+`queue.checkpoint.interval`:: Sets the interval in milliseconds when a checkpoint is forced on the head page.
+Default is `1000`. Set to `0` to eliminate periodic checkpoints.
+
+[[pq-config-notes]]
+==== Configuration notes
+
+Every situation and environment is different, and the "ideal" configuration varies.
+If you optimize for performance, you may increase your risk of losing data.
+If you optimize for data protection, you may impact performance. 
+
+[[pq-size]]
+===== Queue size
+You can control queue size with the `queue.max_events` and  `queue.max_bytes` settings.
+If both settings are specified, Logstash uses whichever criteria is reached
+first. 
+See <<backpressure-persistent-queue>> for behavior when queue limits are
+reached.
+
+[[pq-lower-max_bytes]]
+===== Smaller queue size
+If you are using persistent queues to protect against data loss, but don't
+require much buffering, you can set `queue.max_bytes` to a smaller value, such
+as 10mb.
+A smaller value produces smaller queues and improves queue performance.
+
+*Sample configuration*
+
+[source, yaml]
+-----
+queue.type: persisted
+queue.max_bytes: 10mb 
+-----
+
+[[pq-fewer-checkpoints]]
+===== Fewer checkpoints
+
+Setting `queue.checkpoint.writes` and `queue.checkpoint.acks` to `0` may
+yield maximum performance, but may have potential impact on durability.
+
+In a situation where Logstash is terminated or there is a hardware-level
+failure, any data that has not been checkpointed, is lost. 
+See <<durability-persistent-queues>> to better understand the trade-offs.
+
+[[troubleshooting-pqs]]
+==== Troubleshooting persistent queues
+
+If you see an error message similar to this one, you likely have a persistent queue problem.
```

Comment:
Rather than directly telling users if they see a type of error, shouldn't we first describe what problems can be seen when using the PQ?

For example:

Sometimes, when using persistent queues, these can get corruption and result in the stop of the ingestion delays. [...] continue to explain potential errors [...]. Persistent queue issues can be identified by the following error messages:

Reply:
I tend to agree with this. 

Symptoms of PQ errors can include pipelines, or logstash not being able to start, with the error message shown in this sample

Reply:
Other symptoms to include?

Reply:
They are the main ones, for this particular case anyway